lib/shared/config-manager/src/index.ts

```diff
@@ -186,10 +309,48 @@ export class EnvironmentConfigManager {
                 `Failed to download config, using cached version. url: ${url}.`,
             )
         } else if (responseError?.status === 403) {
-            this.stopPolling()
+            this.cleanup()
             throw new UserError(`Invalid SDK key provided: ${this.sdkKey}`)
         } else {
             throw new Error('Failed to download DevCycle config.')
         }
     }
+
+    private isLastModifiedHeaderOld(lastModifiedHeader: string | null) {
+        const lastModifiedHeaderDate = lastModifiedHeader
+            ? new Date(lastModifiedHeader)
+            : null
+        const configLastModifiedDate = this.configLastModified
+            ? new Date(this.configLastModified)
+            : null
+
+        return (
+            isValidDate(configLastModifiedDate) &&
+            isValidDate(lastModifiedHeaderDate) &&
+            lastModifiedHeaderDate <= configLastModifiedDate
+        )
+    }
+
+    private handleSSEConfig(projectConfig: string) {
+        if (this.enableRealtimeUpdates) {
+            const configBody = JSON.parse(projectConfig) as ConfigBody<string>
+            const originalConfigSSE = this.configSSE
+            this.configSSE = configBody.sse
+
+            // Reconnect SSE if not first config fetch, and the SSE config has changed
+            if (
+                this.hasConfig &&
```

Comment:
when could hasConfig be false when this is called?

Reply:
oh is this to prevent it from happening on the first fetch?

Reply:
yea